src/libsyncengine/propagation/executor/executorworker.cpp

```diff
@@ -1202,27 +1240,29 @@ void ExecutorWorker::handleMoveOp(SyncOpPtr syncOp, bool &hasError) {
         if (!propagateMoveToDbAndTree(syncOp)) {
             LOGW_SYNCPAL_WARN(_logger, L"Failed to propagate changes in DB or update tree for: "
                                                << SyncName2WStr(syncOp->affectedNode()->name()).c_str());
-            _syncPal->setProgressComplete(relativePath, SyncFileStatus::Error);
+            // _executorExitCode and _executorExitCause are set by the above function
             hasError = true;
             return;
         }
     } else {
         bool exists = false;
         if (!hasRight(syncOp, exists)) {
-            // Ignore operation
-            _syncPal->setProgressComplete(relativePath,
-                                          SyncFileStatus::Ignored); // TODO : do not exclude the item from sync, rollback changes
+            ignored = true;
             return;
         }
 
-        if (!generateMoveJob(syncOp)) {
+        if (!generateMoveJob(syncOp, ignored, bypassProgressComplete)) {
+            // _executorExitCode and _executorExitCause are set by the above function
```

Comment:
If all relevant methods behave the same way, and if the fact that they modify the exit code and the exit cause is documented in a docstring (API), such comment a will become useless.

Reply:
Comments removed and @note added.